src/pre_processor.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file pre_processor.h
+ * @brief Interface for pre processing the input image before passing to
+ * `Inference::InferenceCore`
+ * 
+ * @copyright Copyright (C) 2021  Ashwin Maliampurakal
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SRC_PRE_PROCESSOR_H_
+#define SRC_PRE_PROCESSOR_H_
+
+#include "inference_core.h"
+
+namespace PreProcessing {
+
+/**
+ * @brief A structure of the pre processed image
+ *
+ * After pre processing, the image will be resized and normalised before
+ * being passed to the `Inference::InferenceCore` as a `PreProcessedImage`
+ *
+ */
+typedef struct PreProcessedImage {
```

Comment:
typedef is a legacy convention from C and is now handled implicitly by C++ so is not needed here